ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs

```diff
@@ -0,0 +1,23 @@
+﻿using ClosedXML.Excel;
+using NUnit.Framework;
+using System;
+
+namespace ClosedXML_Tests.Excel.Worksheets
+{
+    [TestFixture]
+    public class XLSheetProtectionTests
+    {
+        [Test]
+        public void TestUnprotectWorksheetWithNoPassword()
+        {
+            using (var stream = TestHelper.GetStreamFromResource(TestHelper.GetResourcePath(@"Misc\SHA512PasswordProtection.xlsx")))
+            using (var wb = new XLWorkbook(stream))
+            {
+                wb.Worksheet("Sheet1").Unprotect();
+
+                // Protected with SHA-512 password - not yet supported.
+                Assert.Throws<ArgumentException>(() => wb.Worksheet("Sheet2").Unprotect());
```

Comment:
`NotSupportedException` or `NotImplementedException` would be more informative.

Reply:
That would be a breaking change. I'd rather spend any effort on actually implementing this algorithm.